api_spec.yaml

```diff
@@ -3811,4 +4014,74 @@ components:
     requestUnauthorised:
       type: string
       enum:
-        - Invalid API key
\ No newline at end of file
+        - Invalid API key
+
+    supertokensTenantId:
+      type: string
+      example: 'supertokens'
+
+    thirdPartyProviderConfig:
+      type: object
+      required:
+        - clients
+      properties:
+        name:
+          type: string
+        clients:
+          type: array
+          items:
+            type: object
+            properties:
+              clientType:
+                type: string
+              clientId:
+                type: string
+              clientSecret:
+                type: string
+              scope:
+                type: array
+                items:
+                  type: string
+              additionalConfig:
+                type: object
+        authorizationEndpoint:
+          type: string
+        authorizationEndpointQueryParams:
+          type: object
+        tokenEndpoint:
+          type: string
+        tokenEndpointBodyParams:
+          type: object
+        forcePKCE:
+          type: boolean
+        userInfoEndpoint:
+          type: string
+        userInfoEndpointQueryParams:
+          type: object
+        userInfoEndpointHeaders:
+          type: object
+        jwksURI:
+          type: string
+        oidcDiscoveryEndpoint:
+          type: string
```

Comment:
add a new API for listing all recipe level configs (everything) with tenantId as an input

Reply:
similar to /loginmethods API for the frontend

Reply:
Done, + APIs to create/update tenant and remove tenant